pkg/common/patterns.go

```diff
@@ -37,3 +42,23 @@ func RangeValidation(rangeInput string) bool {
 func ToUpperCase(input string) string {
 	return strings.ToUpper(input)
 }
+
+func (r RegexState) Matches(data []byte) []string {
+	matches := r.compiledRegex.FindAllStringSubmatch(string(data), -1)
+
+	var res []string
+
+	// trim off spaces and different quote types ('").
```

Comment:
Nit: we can just alter them in place here to avoid extra allocs. If you are going to allocate a new slice, pre set the length to the number of matches.